yt_dlp/extractor/facebook.py

```diff
@@ -481,6 +484,28 @@ def _extract_from_url(self, url, video_id):
         webpage = self._download_webpage(
             url.replace('://m.facebook.com/', '://www.facebook.com/'), video_id)
 
+        sjs_data = [self._parse_json(j, video_id, fatal=False) for j in re.findall(
+            r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]
+
+        cookies = self._get_cookies(url)
+        if all(k in cookies for k in ["c_user", "xs"]):
+            if get_first(sjs_data, (
+                    'require', ..., ..., ..., '__bbox', 'define',
+                    lambda _, v: 'CurrentUserInitialData' in v, ..., 'ACCOUNT_ID'), default="0") == "0":
+                raise ExtractorError('Failed to login with provided data.', expected=True)
```

Comment:
Can you explain under what circumstances we would hit this code path?

From what I gather, I don't think we should raise for this, at least not here. We could be extracting a public video and the user doesn't care whether they have successfully authorized with FB or not. Maybe we could raise with this error at the end of the method if no formats can be extracted?

Reply:
This happens when user provided invalid/expired cookies. I think it's important to clearly state that the option they explicitly provided will not take effect.

Reply:
Many users put `--cookies-from-browser` in their config and forget about. Raising is not appropriate. We could warn instead

Reply:
Mm that seems a bit inconsistent to me. Since `_perform_login` is also set to throw whenever the login action fails. And that has been there for a while.
```
            if re.search(r'<form(.*)name="login"(.*)</form>', login_results) is not None:
                error = self._html_search_regex(
                    r'(?s)<div[^>]+class=(["\']).*?login_error_box.*?\1[^>]*><div[^>]*>.*?</div><div[^>]*>(?P<error>.+?)</div>',
                    login_results, 'login error', default=None, group='error')
                if error:
                    raise ExtractorError('Unable to login: %s' % error, expected=True)
```
How is that different to setting user and pass in config then?

Reply:
Is there an approach/flag that can make things like that `strict` if need be? For people that use `yt-dlp` within python, to avoid parsing stdout then?

Reply:
To be fair, cookies are also domain-based at this point.

Reply:
If you can extract the video, then why does it matter cookies match? Do you have any actual use-case?

Reply:
Yes. For a case where you can't extract information whether you're logged in or not. For example, when video is no longer streaming/has been deleted, you will see a banner saying "Content is currently not available"
![image](https://github.com/yt-dlp/yt-dlp/assets/18267226/789f18b7-eb1f-435c-9c41-2f19c54cd943)

When not logged in it will end with a message (most likely) `Cannot parse data...` because it will not always show the `404` equivalent first. 

So, to make sense whether it's a bug in application or you simply didn't log in for the original message to appear, you need to parse the whole context.

Reply:
Ok

Reply:
Changed